ts/components/ui/ModulePaymentNotice.tsx

```diff
@@ -0,0 +1,250 @@
+  interpolate,
+  useAnimatedStyle,
+  useDerivedValue,
+  useSharedValue,
+  withSpring
+} from "react-native-reanimated";
+import {
+  Icon,
+  IOListItemVisualParams,
+  VSpacer
+} from "@pagopa/io-app-design-system";
+import Placeholder from "rn-placeholder";
+import { IOStyles } from "../core/variables/IOStyles";
+import { IOColors } from "../core/variables/IOColors";
+import { IOSpringValues, IOScaleValues } from "../core/variables/IOAnimations";
+import { LabelSmall } from "../core/typography/LabelSmall";
+import { WithTestID } from "../../types/WithTestID";
+import { NewH6 } from "../core/typography/NewH6";
+import { getAccessibleAmountText } from "../../utils/accessibility";
+import { IOBadge } from "../core/IOBadge";
+import I18n from "../../i18n";
+
+export type PaymentNoticeStatus =
+  | "default"
+  | "payed"
+  | "error"
+  | "expired"
+  | "revoked"
+  | "canceled";
+
+type Props = WithTestID<
+  {
+    isLoading?: boolean;
+    accessibilityLabel?: string;
+    title?: string;
+    subtitle: string;
+    onPress: (event: GestureResponderEvent) => void;
+  } & (
+    | {
+        paymentNoticeStatus: Extract<PaymentNoticeStatus, "default">;
+        paymentNoticeAmount: string;
+      }
+    | {
+        paymentNoticeStatus: Exclude<PaymentNoticeStatus, "default">;
+        paymentNoticeAmount?: never;
+      }
+  )
+>;
+
+const ModulePaymentNoticeContent = ({
+  title,
+  subtitle,
+  paymentNoticeStatus,
+  paymentNoticeAmount
+}: Omit<Props, "isLoading" | "onPress" | "testID">) => {
+  const AmountOrBadgeComponent = () => {
+    switch (paymentNoticeStatus) {
+      case "default":
+        return (
+          <NewH6
+            accessibilityLabel={getAccessibleAmountText(paymentNoticeAmount)}
+            color="blue"
+          >
+            {paymentNoticeAmount}
+          </NewH6>
+        );
+      case "payed":
+        return (
+          <IOBadge
+            variant="solid"
+            color="aqua"
+            text={I18n.t("global.modules.paymentNotice.badges.payed")}
+          />
+        );
+      case "error":
+        return (
+          <IOBadge
+            variant="outline"
+            color="red"
+            text={I18n.t("global.modules.paymentNotice.badges.error")}
+          />
+        );
+      case "expired":
+        return (
+          <IOBadge
+            variant="outline"
+            color="grey"
+            text={I18n.t("global.modules.paymentNotice.badges.expired")}
+          />
+        );
+      case "revoked":
+        return (
+          <IOBadge
+            variant="outline"
+            color="grey"
+            text={I18n.t("global.modules.paymentNotice.badges.revoked")}
+          />
+        );
+      case "canceled":
+        return (
+          <IOBadge
+            variant="outline"
+            color="grey"
+            text={I18n.t("global.modules.paymentNotice.badges.canceled")}
+          />
+        );
+    }
+  };
+
+  return (
+    <>
+      <View style={IOStyles.flex}>
```

Comment:
Just to make sure that the block doesn't overflow (or to make it less likely that it will):
```suggestion
      <View style={{ flexGrow: 1, flexShrink: 1 }}>
```